### DISCUSSION TITLE
Azure OpenAI Provider

### DISCUSSION THREAD

**Discussion by rhyswynn** [Holmeswww/AgentKit#8] [Ideas]
Any plans to add an Azure OpenAI Provider option? Thank you!

**Comment by Holmeswww** [comment]
This should be just a couple of lines of changes in https://github.com/Holmeswww/AgentKit/blob/main/src/agentkit/llm_api/GPT.py

Specifically, follow https://learn.microsoft.com/en-us/azure/ai-services/openai/quickstart?tabs=bash%2Cpython-new&pivots=programming-language-python to change [`client`](https://github.com/Holmeswww/AgentKit/blob/df43b1edecac1b8971f5b03f89dc44ad549ce9cc/src/agentkit/llm_api/GPT.py#L24C1-L24C7) to

```
from openai import AzureOpenAI
    
client = AzureOpenAI(
    api_key=os.getenv("AZURE_OPENAI_API_KEY"),  
    api_version="2024-02-01",
    azure_endpoint = os.getenv("AZURE_OPENAI_ENDPOINT")
    )
```

Currently, we don't have plans of officially supporting this feature since the LLM-API is just for a proof-of-concept and could be easily replaced by the users' implementations for other LLMs.

**Comment by rhyswynn** [comment]
Thank you very much for this! I saw the recent update so I thought maybe there were plans to maintain it. I really appreciate all the quick and helpful feedback, and the great work on this tool!

**Comment by Holmeswww** [comment]
Thank you for the active discussion, and apologies for this part of the code not being extended.

**Comment by Holmeswww** [comment]
Thanks to @mattalford, this feature has been merged to the main branch:
6e0c3d0
